src/features/canvass/components/GLCanvassMap/index.tsx

```diff
@@ -182,6 +223,44 @@ const GLCanvassMap: FC<Props> = ({ assignment, selectedArea }) => {
     }
   };
 
+  const locateUser = () => {
+    if (!shouldLocate) {
+      setUserLocation(null);
+      return;
+    }
+
+    setTimeout(() => {
+      navigator.geolocation.getCurrentPosition(
+        (pos) => {
+          const lat = pos.coords.latitude as Latitude;
+          const lng = pos.coords.longitude as Longitude;
+          setUserLocation([lng, lat]);
+          setUserAccuracy(pos.coords.accuracy ?? null);
+
+          locateUser();
+        },
+        null,
+        { enableHighAccuracy: true, timeout: 5000 }
+      );
+    }, RELOCATION_TIMEOUT);
+  };
```

Comment:
How about instead of a timer we could use [`Geolocation.watchPosition()`](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/watchPosition)?